### PR TITLE
Feat: activity page contributions, part 1

### PR DESCRIPTION
**Pull request by wingfeatherwave**
Screenshot 2023-08-22 at 3 51 20 PM

Test: 
* set local storage feature flag in browser console
`localStorage.setItem('feature:activityContributions', true);`
* add a contribution on the circle page

up next:
* edit contributions that are in current or future epoch
* delete contributions

### <samp>🤖 Generated by Copilot at 7d006f3</samp>

This pull request adds a feature to enable users to add contributions to their circle activities. It introduces a new `ContributionForm` component that handles the input, preview, and saving of the contribution data. It also updates the `CircleActivityPage` component to render the form based on a feature flag. Additionally, it improves the data fetching and refreshing logic for the activity list component.

### <samp>🤖 Generated by Copilot at 7d006f3</samp>

> _`ContributionForm`_
> _A new feature for circles_
> _Autumn of sharing_

### <samp>🤖 Generated by Copilot at 7d006f3</samp>

*  Add a new feature flag for 'activityContributions' and a function to check it ([link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-5c118d9192f4171e7f6592b2205521832975d0666a8673815188b73c9644088aR4))
*  Create a `ContributionForm` component to allow users to enter and submit contributions in markdown format ([link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60R1-R209))
*  Conditionally render the `ContributionForm` component in the `CircleActivityPage` component if the feature flag is enabled ([link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-ad4502b0dfe7c31d2c3282e694732d7e7e4d5095133d8274e63f50746492385dL6-R9), [link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-ad4502b0dfe7c31d2c3282e694732d7e7e4d5095133d8274e63f50746492385dR28))
*  Enable periodic and window focus refetching of activity data in the `useInfiniteActivities` hook ([link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-de71c4ba5014d75c0e22b56c2dea640e7a20fda5614d3928efcc786e8b663391L90-R91))
*  Add a temporary console log statement for debugging the saving status of contributions in the `ContributionsPage` component ([link](https://github.com/coordinape/coordinape/pull/2301/files?diff=unified&w=0#diff-93b9497dda713d9f747bd53334a4e8e9f2280f0b65fc9646f29daa511cdf0ab5R308-R309))

### CODE REVIEW

File: src/config/features.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // add your own feature names here
 
 export type FeatureName =
+  | 'activityContributions'
```
Comment (crabsinger) [inline src/config/features.ts:4]: previous pattern is `_` not camelcase it seems?

File: src/pages/ContributionsPage/ContributionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+  const { mutate: createContribution, reset: resetCreateMutation } =
+    useMutation(createContributionMutation, {
+      onSuccess: async newContribution => {
+        queryClient.invalidateQueries({
+          queryKey: [QUERY_KEY_ALLOCATE_CONTRIBUTIONS],
+        });
+        setTimeout(
+          () => queryClient.invalidateQueries(ACTIVITIES_QUERY_KEY),
+          1000
```
Comment (crabsinger) [inline src/pages/ContributionsPage/ContributionForm.tsx:62]: @CryptoGraffe @wingfeatherwave these seems a bit hacky... wait 1 sec then refetch... might be worth thinking if we can find a more reliable solution pre-public launch.

File: src/pages/ContributionsPage/ContributionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                  }}
+                >
+                  Markdown Supported
+                </Text>
+              </Box>
+            )}
+            <Button
+              color="cta"
+              onClick={() => saveContribution(descriptionField.value)}
+            >
```
Comment (crabsinger) [inline src/pages/ContributionsPage/ContributionForm.tsx:201]: If I double click the Add Contribution button I get an empty contribution so think we should add
```suggestion
              disabled={!descriptionField.value}
            >
```

**Review by crabsinger — COMMENTED** [review]
two small changes before shippable, looks nice!

File: src/pages/ContributionsPage/ContributionForm.tsx (hunk 2 above)
Comment (CryptoGraffe) [reply src/pages/ContributionsPage/ContributionForm.tsx:62]: The way to do that would be to make the activity insert inline with contribution creation and return it when creating contribution

**Review by crabsinger — APPROVED** [review]

**Pull request merged**